src/Features/Core/Portable/PullMemberUp/PullMemberUpAnalyzer.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.  
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+
+namespace Microsoft.CodeAnalysis.CodeRefactorings.PullMemberUp
+{
+    internal class PullMembersUpAnalysisBuilder
+    {
+        internal static AnalysisResult BuildAnalysisResult(
+            INamedTypeSymbol targetSymbol,
+            IEnumerable<ISymbol> members)
+        {
+            var memberResult = members.Select(member =>
+            {
+                if (targetSymbol.TypeKind == TypeKind.Interface)
+                {
+                    return new MemberAnalysisResult(
+                        member, member.DeclaredAccessibility != Accessibility.Public,
+                        member.IsStatic);
+                }
+                else
+                {
+                    var changeOriginToPublic = false;
+                    var changeOriginToNonStatic = false;
+                    return new MemberAnalysisResult(member, changeOriginToPublic, changeOriginToNonStatic);
+                }
+            });
+
+            return new AnalysisResult(targetSymbol, memberResult);
+        }
+    }
+
+    internal class MemberAnalysisResult
+    {
+        public readonly ISymbol _member;
+
+        public readonly bool _changeOriginToPublic;
+
+        public readonly bool _changeOriginToNonStatic;
```

Comment:
1. needs document.
2. probably shoudl be called 'ChangeOriginal...'
3. public readonlies should be PascalCased.